src/BaGet.Hosting/Controllers/SearchController.cs

```diff
@@ -43,29 +43,42 @@ public SearchController(ISearchService searchService)
         }
 
         public async Task<ActionResult<AutocompleteResponse>> AutocompleteAsync(
-            [FromQuery(Name = "q")] string query = null,
-            [FromQuery]int skip = 0,
-            [FromQuery]int take = 20,
+            [FromQuery(Name = "q")] string autocompleteQuery = null,
+            [FromQuery(Name = "id")] string versionsQuery = null,
             [FromQuery]bool prerelease = false,
             [FromQuery]string semVerLevel = null,
+            [FromQuery]int skip = 0,
+            [FromQuery]int take = 20,
 
             // These are unofficial parameters
             [FromQuery]string packageType = null,
             CancellationToken cancellationToken = default)
         {
-            // TODO: Support versions autocomplete.
-            // See: https://github.com/loic-sharma/BaGet/issues/291
             var request = new AutocompleteRequest
             {
-                Skip = skip,
-                Take = take,
                 IncludePrerelease = prerelease,
                 IncludeSemVer2 = semVerLevel == "2.0.0",
                 PackageType = packageType,
-                Query = query,
+                Skip = skip,
+                Take = take
             };
 
-            return await _searchService.AutocompleteAsync(request, cancellationToken);
+            // Default to autocomplete, just like nuget.org does
+            if(autocompleteQuery != null)
+            {
+                request.Query = autocompleteQuery;
+
+                return await _searchService.AutocompleteAsync(request, cancellationToken);
+            }
+
+            if(versionsQuery != null)
+            {
+                request.Query = versionsQuery;
+
+                return await _searchService.ListPackageVersionsAsync(request, cancellationToken);
+            }
+
+            return new AutocompleteResponse();
```

Comment:
Your logic is perfectly correct but I'd prefer to rearrange it so that line 81 isn't necessary. What do you think of something like:

```suggestion
            // If only "id" is provided, find package versions. Otherwise, find package ids.
            if (versionsQuery != null && autocompleteQuery == null)
            {
                request.Query = versionsQuery;

                return await _searchService.ListPackageVersionsAsync(request, cancellationToken);
            }
            else
            {
                request.Query = autocompleteQuery;

                return await _searchService.AutocompleteAsync(request, cancellationToken);
            }
```